crypto/eth2_keystore/src/json_keystore/kdf_module.rs

```diff
@@ -54,6 +55,126 @@ impl Kdf {
+                }
+
+                // Validate `salt` length
+                if params.salt.is_empty() {
+                    return Err(Error::InvalidScryptParam);
+                } else if params.salt.len() < SALT_SIZE / 2 {
+                    eprintln!(
+                        "WARN: Salt is too short {}, we recommend {}",
+                        params.salt.len(),
+                        SALT_SIZE
+                    );
+                } else if params.salt.len() > SALT_SIZE * 2 {
+                    eprintln!(
+                        "WARN: Salt is too long {}, we recommend {}",
+                        params.salt.len(),
+                        SALT_SIZE
+                    );
+                }
+
+                Ok(())
+            }
+            Kdf::Scrypt(params) => {
+                // RFC7914 declares that all these parameters must be greater than 1:
+                //
+                // - `N`: costParameter.
+                // - `r`: blockSize.
+                // - `p`: parallelizationParameter
+                //
+                // Reference:
+                //
+                // https://tools.ietf.org/html/rfc7914
+                if params.n <= 1 || params.r == 0 || params.p == 0 {
+                    return Err(Error::InvalidScryptParam);
+                }
+
+                // We always compute a derived key of 32 bytes so reject anything that
+                // says otherwise.
+                if params.dklen != DKLEN {
+                    return Err(Error::InvalidScryptParam);
+                }
+
+                // Ensure that `n` is power of 2.
+                if params.n != 2u32.pow(log2_int(params.n)) {
+                    return Err(Error::InvalidScryptParam);
+                }
+
+                // Maximum Parameters
+                //
+                // Uses a u32 to store value thus maximum memory usage is 4GB.
+                //
+                // Note: Memory requirements = 128*n*p*r
+                let mut npr: u32 = params
+                    .n
+                    .checked_mul(params.p)
+                    .ok_or(Error::InvalidScryptParam)?;
+                npr = npr.checked_mul(params.r).ok_or(Error::InvalidScryptParam)?;
+                npr = npr.checked_mul(128).ok_or(Error::InvalidScryptParam)?;
+
+                // Minimum Parameters
+                let default = Scrypt::default_scrypt(vec![0u8; 32]);
+                let default_npr = 128 * default.n * default.p * default.r;
+                if npr < default_npr {
+                    eprintln!("WARN: Scrypt parameters are too weak (n: {}, p: {}, r: {}), we recommend (n: 262,144, p: 1, r: 8)", params.n, params.p, params.r);
+                }
+
+                // Validate `salt` length
+                if params.salt.is_empty() {
+                    return Err(Error::InvalidScryptParam);
+                } else if params.salt.len() < SALT_SIZE / 2 {
```

Comment:
Nit: we could wrap this check and the next one in a function and use it for pbkdf2 too.

Reply:
Yep can do :)